dali/server/daldap.cpp

```diff
@@ -121,44 +121,46 @@ class CDaliLdapConnection: public CInterface, implements IDaliLdapConnection
     {
         if (!ldapsecurity||((getLDAPflags()&DLF_ENABLED)==0)) 
             return 255;
+
+        StringBuffer username;
+        StringBuffer password;
+        if (udesc)
+        {
+            udesc->getUserName(username);
+            udesc->getPassword(password);
+        }
+        else
+        {
+            WARNLOG("NULL UserDescriptor in daldap.cpp getPermissions('%s')",key ? key : "NULL");
+        }
+
+        if (0 == username.length())
+        {
+            username.append(filesdefaultuser);
+            decrypt(password, filesdefaultpassword);
```

Comment:
default user/pw is optional so this can be NULL. In this case, authentication will fail and the user will be denied access